src/tdvp_step.jl

```diff
@@ -128,7 +129,7 @@ function tdvp_sweep(
   end
   # Just to be sure:
   normalize && normalize!(psi)
-  return psi, PH, TDVPInfo(maxtruncerr)
+  return psi, PH, TDVPInfo(maxtruncerr, converged)
```

Comment:
I don't really see why it is useful to return the `converged` value here. It seems like it only returns the convergence information of the final site update of the sweep, which doesn't seem so helpful.

Reply:
Instead I think it should be passed to the `observer` above (inside the loop) so that the convergence information at each site can be saved.

Reply:
In fact, I think we should return the entire `info` object from `tdvp_site_update!`, not just `converged`, and pass that to the `observer` inside the loop. Users can choose if they only want to save `info.converged` in the Observer.

Reply:
Good point. I've implemented this.

Reply:
Looking through the code, I don't see this implemented. I see that you are now passing `info` to `TDVPInfo`, but I don't think we should be doing that. You should just pass `info` to the `update!(observer; ...)` function in the loop over `sweep_bonds`.

Reply:
I've made a further change to reflect what I think you mean. Basically, I've taken all the convergence info out of TDVPInfo, and don't put it in the step_observer anymore either. Instead, I've put all the convergence info in observer. If this is still not what you intend, perhaps we should choose another channel of communication, and hash this out in more detail.

Reply:
Looks good now, thanks!